resources/aws_ec2.go

```diff
@@ -486,6 +489,68 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsEndpoint := obj.snsServer()
+	// shutdown the sns endpoint when we're done
+	obj.wg.Add(1)
+	go func() {
+		defer obj.wg.Done()
+		select {
+		case <-obj.closeChan:
+			if err := snsEndpoint.Shutdown(nil); err != nil {
+				log.Printf("%s: error stopping sns endpoint: %s", obj, err)
+				return
+			}
+			log.Printf("%s: Stopped SNS Endpoint", obj)
+		}
+	}()
+	obj.wg.Add(1)
+	// start the endpoint
+	go func() {
+		defer obj.wg.Done()
+		defer close(obj.awsChan)
+		if err := snsEndpoint.ListenAndServe(); err != nil {
```

Comment:
If you trace this `ListenAndServe` method, eventually if it doesn't error initially, you eventually get here:

https://golang.org/src/net/http/server.go?s=87921:87976#L2676

You don't always need to trace the code, (starwars: UTSL) because it was mentioned in the docs: ` ListenAndServe always returns a non-nil error.`, but tracing told us why if you follow the code to `Serve`. Fun!

So this *will* error... We probably don't want to always complain if the "error" is a "we shutdown now" error. So possibly some error parsing/detection is needed here.

Reply:
Weird, I guess I didn't notice, since snsWatch has stopped reading from the channel when it shuts down. I suppose if it returns early for some reason, we'll have to deal with it. I'll spend some time in the source and fix.

Reply:
> snsWatch has stopped reading from the channel when it shuts down I suppose if it returns early for some reason, we'll have to deal with it.

Yup. Or I think more likely is that it can't start up due to the port being in use maybe?